### PR TITLE
Remember widget visibility in default view mode

### PR DESCRIPTION
**Pull request by tmoerschell**
Fix for issue #5127
Remove unused code and remember visibility of widgets (menubar, toolbar, sidebar) by saving it to the settings
The changes are also persistent between sessions since they are saved to the settings

### CODE REVIEW

**Review by bhennion — CHANGES_REQUESTED** [review]
Thanks for the PR and for fixing this regression. I have a couple of comments below.
Also, the class FullscreenHandler is really not useful anymore (it has only one member variable `bool fullscreen` whose getter is only used in Control::isFullscreen() which in turn is never used).
I think you could simply remove FullscreenHandler entirely and implement a MainWindow::goFullscreen(bool) calling the gtk functions.

File: src/core/control/settings/Settings.cpp [hunk 1]
```diff
@@ -2007,7 +2007,10 @@ auto Settings::getButtonConfig(unsigned int id) -> ButtonConfig* {
     return this->buttonConfig[id].get();
 }
 
-void Settings::setViewMode(ViewModeId mode, ViewMode viewMode) { viewModes.at(mode) = viewMode; }
+void Settings::setViewMode(ViewModeId mode, ViewMode viewMode) {
+    viewModes.at(mode) = viewMode;
+    save();
+}
```
[the remark this answers is not in the document]
Comment (tmoerschell) [reply src/core/control/settings/Settings.cpp]: Fixed in [15b2786](https://github.com/xournalpp/xournalpp/pull/5130/commits/15b2786fab9e7ea2159228344e64441c56b5a812)

File: src/core/gui/MainWindow.cpp [hunk 2]
```diff
@@ -369,18 +377,32 @@ void MainWindow::dragDataRecived(GtkWidget* widget, GdkDragContext* dragContext,
...
 void MainWindow::viewShowSidebar(GtkCheckMenuItem* checkmenuitem, MainWindow* win) {
-    bool a = gtk_check_menu_item_get_active(checkmenuitem);
-    if (win->control->getSettings()->isSidebarVisible() == a) {
+    bool showSidebar = gtk_check_menu_item_get_active(checkmenuitem);
+    Settings* settings = win->control->getSettings();
+    if (settings->isSidebarVisible() == showSidebar) {
         return;
     }
-    win->setSidebarVisible(a);
+    if (!settings->isFullscreen() && !settings->isPresentationMode()) {
```
[the remark this answers is not in the document]
Comment (tmoerschell) [reply src/core/gui/MainWindow.cpp]: Done in [4bb4d44](https://github.com/xournalpp/xournalpp/pull/5130/commits/4bb4d4466e19e2cbb853fcf8c36fedb478a743a7)

File: src/core/gui/MainWindow.cpp [hunk 3]
```diff
@@ -448,7 +470,16 @@ void MainWindow::updateScrollbarSidebarPosition() {
     g_object_unref(boxContents);
 }
 
-void MainWindow::buttonCloseSidebarClicked(GtkButton* button, MainWindow* win) { win->setSidebarVisible(false); }
+void MainWindow::buttonCloseSidebarClicked(GtkButton* button, MainWindow* win) {
+    Settings* settings = win->control->getSettings();
+    if (!settings->isFullscreen() && !settings->isPresentationMode()) {
```
[the remark this answers is not in the document]
Comment (tmoerschell) [reply src/core/gui/MainWindow.cpp]: Done in [4bb4d44](https://github.com/xournalpp/xournalpp/pull/5130/commits/4bb4d4466e19e2cbb853fcf8c36fedb478a743a7)
There was also a 4th occurence on line 191

**Comment by tmoerschell** [comment]
I put the gtk functions into Control::setFullscreen for now. Do you think it would make sense to create an additional function for that in the MainWindow class? If so I'd rather call it MainWindow::setFullscreen(bool enable)

**Comment by bhennion** [comment]
> I put the gtk functions into Control::setFullscreen for now. Do you think it would make sense to create an additional function for that in the MainWindow class? If so I'd rather call it MainWindow::setFullscreen(bool enable)

Yes I would. The underlying idea is that the MainWindow should be a wrapper around a GtkWindow, and any access to this GtkWindow should go through this wrapper. This is far from being the case right now, but this is some guideline you can keep in mind.

**Comment by tmoerschell** [comment]
Ok will do, but I'm unsure about the name now: 
`setFullscreen(bool)` looks like a setter for a class member, which isn't the case. 
`goFullscreen(bool)` doesn't sound good for disabling fullscreen. 
`switchFullscreen(bool)` ? Just `fullscreen(bool)` ?
What do you think would fit best?

**Comment by bhennion** [comment]
> What do you think would fit best?

I guess `setFullscreen(bool)` look like the most natural to me. That's pretty subjective though, so pick the one you prefer.

**Comment by tmoerschell** [comment]
Ok I found ´toggleFullscreen´ would fit well. Sorry for the delay, everything should be good now

**Review by bhennion — APPROVED** [review]
LGTM! Thanks for the PR!
(Squashing and) Merging in 24 hours unless an objection is raised.

**Pull request merged**